### PR TITLE
[POAE7-2745] Add Target Information to LLVMJITModule

### PR DESCRIPTION
**Pull request by bigPYJ1151**
### What changes were proposed in this pull request?

Add Target Information to LLVMJITModule

### Why are the changes needed?

For IR Optimization

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unnecessary

### Which label does this PR belong to?

INFRA

### CODE REVIEW

File: src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITEngine.cpp [hunk 1]
```diff
@@ -152,7 +62,7 @@ std::unique_ptr<LLVMJITEngine> LLVMJITEngineBuilder::build() {
       .setErrorStr(&error);
 
   auto engine = std::make_unique<LLVMJITEngine>();
-  engine->engine = eb.create(buildTargetMachine());
+  engine->engine = eb.create(tm_.release());
```
Comment (xieqi) [inline src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITEngine.cpp:65]: why tm_ as unique ptr?

Comment (bigPYJ1151) [reply src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITEngine.cpp:65]: Just to make ownership clear.

Comment (xieqi) [reply src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITEngine.cpp:65]: who has the ownership after release?

Comment (bigPYJ1151) [reply src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITEngine.cpp:65]: image
```llvm::EngineBuilder``` will take it.

**Review by xieqi — APPROVED** [review]

**Review by kd97 — APPROVED** [review]

**Pull request merged**